src/test/rpc/AccountTx_test.cpp

```diff
@@ -120,22 +120,37 @@ class AccountTx_test : public beast::unit_test::suite
         // All other ledgers have no txs
 
         auto hasTxs = [apiVersion](Json::Value const& j) {
-            return j.isMember(jss::result) &&
-                (j[jss::result][jss::status] == "success") &&
-                (j[jss::result][jss::transactions].size() == 2) &&
-                (j[jss::result][jss::transactions][0u][jss::tx]
-                  [jss::TransactionType] == jss::AccountSet) &&
-                (j[jss::result][jss::transactions][1u][jss::tx]
-                  [jss::TransactionType] == jss::Payment) &&
-                (j[jss::result][jss::transactions][1u][jss::tx]
-                  [jss::DeliverMax] == "10000000010") &&
-                ((apiVersion > 1 &&
-                  !j[jss::result][jss::transactions][1u][jss::tx].isMember(
-                      jss::Amount)) ||
-                 (apiVersion <= 1 &&
-                  j[jss::result][jss::transactions][1u][jss::tx][jss::Amount] ==
-                      j[jss::result][jss::transactions][1u][jss::tx]
-                       [jss::DeliverMax]));
+            switch (apiVersion)
+            {
+                case 1:
+                    return j.isMember(jss::result) &&
+                        (j[jss::result][jss::status] == "success") &&
+                        (j[jss::result][jss::transactions].size() == 2) &&
+                        (j[jss::result][jss::transactions][0u][jss::tx]
+                          [jss::TransactionType] == jss::AccountSet) &&
+                        (j[jss::result][jss::transactions][1u][jss::tx]
+                          [jss::TransactionType] == jss::Payment) &&
+                        (j[jss::result][jss::transactions][1u][jss::tx]
+                          [jss::DeliverMax] == "10000000010") &&
+                        (j[jss::result][jss::transactions][1u][jss::tx]
+                          [jss::Amount] ==
+                         j[jss::result][jss::transactions][1u][jss::tx]
+                          [jss::DeliverMax]);
+                case 2:
+                    return j.isMember(jss::result) &&
+                        (j[jss::result][jss::status] == "success") &&
+                        (j[jss::result][jss::transactions].size() == 2) &&
+                        (j[jss::result][jss::transactions][0u][jss::tx_json]
+                          [jss::TransactionType] == jss::AccountSet) &&
+                        (j[jss::result][jss::transactions][1u][jss::tx_json]
+                          [jss::TransactionType] == jss::Payment) &&
+                        (j[jss::result][jss::transactions][1u][jss::tx_json]
+                          [jss::DeliverMax] == "10000000010") &&
+                        (!j[jss::result][jss::transactions][1u][jss::tx_json]
+                              .isMember(jss::Amount));
+                default:
```

Comment:
I may be wrong here, but what happens if v3 comes in and we don't do any specific change to this command or test. Should it not still follow the logic of case 2 ?

Reply:
I think we can coalesce multiple cases:
```
case 2:
case 3:
// existing tests
```

Reply:
Would it not be better to follow this pattern ? (again, i'm not sure if switch case statements makes more sense...)
```
if (version < FEATURE_X_VERSION) {
  // old behavior
} else {
  // new behavior
}

if (version >= FEATURE_X_VERSION) {
  // new behavior where old behavior was just "do nothing"
}

if (version < FEATURE_X_VERSION) {
  // old behavior where new behavior is "stop doing that"
}
```

In this particular case just:

```
if (version < 2u) {
  // old behavior
} else {
  // new behavior
}
```

Reply:
not the biggest deal in any case 😆

Reply:
compilers rather good at catching bugs which arise from missing a `case` in a `switch` statement, and very bad at catching bugs from bad logic in `if ... else if .. else` series of statements. This is why I prefer using `switch`. As soon as we add version 3 and start testing it, this will fail (due to `default: return false;`) which is exactly as we want it, so we know to make adjustment. If these adjustments become a problem, we can change style, but I hope by then we will have a better way of testing different API versions (I aim to work on it, and `forAllApiVersions` is a good first effort).

Reply:
Thanks for the reply, that is good to know.